mediapipe/tasks/ios/test/vision/image_classifier/ImageClassifierTests.swift

```diff
@@ -0,0 +1,690 @@
+    XCTAssertEqual(
+      imageClassifierResult.classificationResult.classifications.count, 
+      1)
+    XCTAssertEqual(
+      imageClassifierResult.classificationResult.classifications[0].headIndex, 
+      0)
+  }
+
+  func imageWithFileInfo(_ fileInfo: FileInfo) throws -> MPImage {
+    let mpImage = try XCTUnwrap(
+      MPImage.imageFromBundle(
+        withClass: type(of: self),
+        filename: fileInfo.name,
+        type: fileInfo.type))
+
+   return mpImage
+  }
+
+  func imageClassifierOptionsWithModelPath(
+    _ modelPath: String?
+  ) throws -> ImageClassifierOptions {
+    let modelPath = try XCTUnwrap(modelPath)
+
+    let imageClassifierOptions = ImageClassifierOptions()
+    imageClassifierOptions.baseOptions.modelAssetPath = modelPath
+
+    return imageClassifierOptions
+  }
+
+  func assertCreateImageClassifierThrowsError(
+    imageClassifierOptions: ImageClassifierOptions,
+    expectedErrorDescription: String
+  ) {
+    do {
+      let imageClassifier = try ImageClassifier(options: imageClassifierOptions)
+      XCTAssertNil(imageClassifier)
+    } catch {
+      assertEqualErrorDescriptions(
+        error,
+        expectedLocalizedDescription: expectedErrorDescription)
+    }
+  }
+
+  func assertImageClassifierResult(
+    _ imageClassifierResult: ImageClassifierResult,
+    hasCategoryCount expectedCategoryCount: Int,
+    andCategories expectedCategories: [ResultCategory]
+  ) throws {
+    assertImageClassifierResultHasOneHead(imageClassifierResult)
+    let categories = 
+      imageClassifierResult.classificationResult.classifications[0].categories
+
+    XCTAssertEqual(categories.count, expectedCategoryCount)    
+    assertEqualCategoryArrays(
+      categoryArray:
+        Array(categories.prefix(expectedCategories.count)),
+      expectedCategoryArray: expectedCategories)
+  }
+
+  func assertResultsForClassifyImage(
+    _ image: MPImage,
+    usingImageClassifier imageClassifier: ImageClassifier,
+    hasCategoryCount expectedCategoryCount: Int,
+    andCategories expectedCategories: [ResultCategory]
+  ) throws {
+    let imageClassifierResult =
+      try XCTUnwrap(
+        imageClassifier.classify(image: image))
+    
+    try assertImageClassifierResult(
+      imageClassifierResult,
+      hasCategoryCount: expectedCategoryCount,
+      andCategories: expectedCategories
+    )
+  }
+
+  func assertResultsForClassifyImageWithFileInfo(
+    _ fileInfo: FileInfo,
+    usingImageClassifier imageClassifier: ImageClassifier,
+    hasCategoryCount expectedCategoryCount: Int,
+    andCategories expectedCategories: [ResultCategory]
+  ) throws {
+    let mpImage = try XCTUnwrap(
+      imageWithFileInfo(fileInfo))
+
+    try assertResultsForClassifyImage(
+      mpImage,
+      usingImageClassifier: imageClassifier,
+      hasCategoryCount: expectedCategoryCount,
+      andCategories: expectedCategories
+    )
+  }
+
+  func testCreateImageClassifierWithCategoryAllowlistAndDenylistFails() throws {
```

Comment:
This seems to be missing `testCreateImageClassifierWithMissingModelPathFails`

Reply:
That scenario will never happen in Swift and hence can't be tested. `modelPath` param is strictly `nonnull` and hence optionals cannot be passed to it.  Since Swift is strongly typed, If we try to instantiate a non optional path which doesn't point to any existing file using the native apis, an error is raised. 
Objective C only raises a warning if you pass nil to non null param. Hence this test was needed.